attachment.go

```diff
@@ -29,6 +29,11 @@ func (c *Client) UploadAttachment(ctx context.Context, addrKR *crypto.KeyRing, r
 		return Attachment{}, fmt.Errorf("failed to encrypt attachment: %w", err)
 	}
 
+	// Sanitize content disposition
+	if req.ContentID == "" || (req.Disposition != AttachmentDisposition && req.Disposition != InlineDisposition) {
+		req.Disposition = AttachmentDisposition
+	}
+
```

Comment:
I'd prefer we didn't perform any sanitization here -- as an API library, we shouldn't secretly modify the request body during upload like this, otherwise users of the library will be confused why their request bodies are being changed.

Can you please perform the sanitization [here](https://github.com/ProtonMail/proton-bridge/blob/v3/internal/user/smtp.go#L390) instead?

Reply:
> if I want to reuse this function in some import export tool I have to sanitise again in that code base

Yes, and you should. It's not stated anywhere that `disposition` will magically be set to `attachment` in some cases; a user of the library shouldn't throw "invalid" data at the API hoping that it magically fixes it. And in future, the API might be updated to support additional dispositions; callers shouldn't then be surprised when their newly supported dispositions are being discarded by the library.

Reply:
so what about now? 

I don't want to send requests which we know they are going to fail, should we?

Reply:
I still disagree with this... what's the value? The API will already catch these errors and generate sentry reports; by returning an error early ourselves, the outcome is the same (the request fails), but we lose information. And the API might one day be upgraded to support a wider range of attachments; we shouldn't then have to remove these additional checks.

Reply:
`go-proton-api` is meant to be thin; if the caller knows a request should fail, then the caller shouldn't construct an invalid request. And if they do that anyway, then there's a caller bug, and it will be reported via sentry.

Reply:
My intention was to have less maintenance in future by not having same code on multiple places and second attempt was to discover the issue before reaching the production and don't rely on fact it's gonna be reported to sentry.

Your points are valid too so I removed it completely :)

Reply:
Thanks :)